lib/chef/provider/service/macosx.rb

```diff
@@ -77,6 +95,11 @@ def define_resource_requirements
               @current_resource.running(false)
             end
           end
+
+          requirements.assert(:all_actions) do |a|
+            a.assertion { @plist.nil? ? true : ::File.exists?(@plist) }
+            a.failure_message Chef::Exceptions::Service, "#{@plist} does not exist"
+          end
```

Comment:
i dont know, im getting confused. In my test case it worked. Im not really sure how he got this error